### DISCUSSION TITLE
Can't make Google Oauth work

### DISCUSSION THREAD

**Discussion by nick-fullpath** [open-webui/open-webui#5345] [General]
# Bug Report

## Installation Method
Running the project as a docket container on a VM in GCP.
The access is via ALB with SSL and a custom DNS "opengpt.********.com"

## Environment

- **Open WebUI Version:** v0.3.2

- **Operating System:** [Docker container running on Ubuntu 20.04]

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [X] I have included the Docker container logs.
- [X] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
After Creating a new Oauth Credentials on GCP and Passing the environment variables required to google auth connection which are "WEBUI_URL", "GOOGLE_CLIENT_ID", "GOOGLE_CLIENT_SECRET" i should be able to see option to connect with google oauth to the app.

## Actual Behavior:

No Oauth with google connection option present, just the regular login with username and password

## Description

**Bug Summary:**
After Creating a new Oauth Credentials on GCP and Passing the environment variables required to google auth connection which are "WEBUI_URL", "GOOGLE_CLIENT_ID", "GOOGLE_CLIENT_SECRET" i should be able to see option to connect with google oauth to the app.

Unfortunately i see only the regular username and password login option.

## Reproduction Details

**Steps to Reproduce:**
1. Create a google oauth credentials on GCP with my redirection url as the documentation advise "https://opengpt.********.com/oauth/google/callback", Copy my Client ID and Client Secret.
2. Run the Docker container of open-webui with the following command:
`docker run -d -p 8080:8080 -e OPENAI_API_KEY=******* -e GOOGLE_CLIENT_ID=********.apps.googleusercontent.com -e GOOGLE_CLIENT_SECRET=******** -e WEBUI_URL=https://opengpt.******.com -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`
3. When the container finish its loading and im navigating to "https://opengpt.******.com" i'm still seeing the regular login screen with username and password.

## Logs and Screenshots

**Docker Container Logs:**
`Loading WEBUI_SECRET_KEY from file, not provided as an environment variable.
Generating WEBUI_SECRET_KEY
Loading WEBUI_SECRET_KEY from .webui_secret_key
INFO:     Started server process [1]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
/app

  ___                    __        __   _     _   _ ___ 
 / _ \ _ __   ___ _ __   \ \      / /__| |__ | | | |_ _|
| | | | '_ \ / _ \ '_ \   \ \ /\ / / _ \ '_ \| | | || | 
| |_| | |_) |  __/ | | |   \ V  V /  __/ |_) | |_| || | 
 \___/| .__/ \___|_| |_|    \_/\_/ \___|_.__/ \___/|___|
      |_|                                               

      
v0.3.2 - building the best open-source AI user interface.

https://github.com/open-webui/open-webui

INFO:apps.openai.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Name or service not known]
INFO:     127.0.0.1:57920 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:51014 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:55044 - "GET /health HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYGo- HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /api/config HTTP/1.1" 200 OK
INFO:     147.236.177.170:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYGtN HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYGwq HTTP/1.1" 200 OK
connect  9ugdc5aVB8XVx-m0AAAD
INFO:     147.236.177.170:0 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P7WYGw8&sid=l5Qd0xtv3gk4fZG7AAAB HTTP/1.1" 200 OK
INFO:     147.236.177.170:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYGw9&sid=l5Qd0xtv3gk4fZG7AAAB HTTP/1.1" 200 OK
INFO:     ('147.236.177.170', 0) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=l5Qd0xtv3gk4fZG7AAAB" [accepted]
INFO:     connection open
INFO:     147.236.177.170:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYGyv&sid=l5Qd0xtv3gk4fZG7AAAB HTTP/1.1" 200 OK
INFO:     ('147.235.193.174', 0) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=G0HR9_adKiQ3wYrvAAAC" [accepted]
INFO:     connection open
INFO:     147.235.193.174:0 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P7WYH0F&sid=G0HR9_adKiQ3wYrvAAAC HTTP/1.1" 200 OK
INFO:     147.236.177.170:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYG_a&sid=l5Qd0xtv3gk4fZG7AAAB HTTP/1.1" 200 OK
INFO:     connection closed
INFO:     2.53.174.122:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHBa HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHFZ HTTP/1.1" 200 OK
connect  hn1bHf9RdUl4KPTaAAAG
INFO:     2.53.174.122:0 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P7WYHFc&sid=OPGFK9cpWSPSllA0AAAE HTTP/1.1" 200 OK
INFO:     2.53.174.122:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHFe&sid=OPGFK9cpWSPSllA0AAAE HTTP/1.1" 200 OK
connect  0du6xhPiHFp5ES0YAAAH
INFO:     147.235.193.174:0 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P7WYHIM&sid=_PriJjHWdTiTM5m7AAAF HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHIN&sid=_PriJjHWdTiTM5m7AAAF HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHIh HTTP/1.1" 200 OK
INFO:     ('147.235.193.174', 0) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=_PriJjHWdTiTM5m7AAAF" [accepted]
INFO:     connection open
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHLF&sid=_PriJjHWdTiTM5m7AAAF HTTP/1.1" 200 OK
connect  tcBl_nBIkruX99QJAAAJ
INFO:     147.235.193.174:0 - "POST /ws/socket.io/?EIO=4&transport=polling&t=P7WYHLW&sid=cOHi1ZjDd2-KZeqyAAAI HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHLW.0&sid=cOHi1ZjDd2-KZeqyAAAI HTTP/1.1" 200 OK
INFO:     ('147.235.193.174', 0) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=cOHi1ZjDd2-KZeqyAAAI" [accepted]
INFO:     connection open
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHNy&sid=_PriJjHWdTiTM5m7AAAF HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHOa&sid=cOHi1ZjDd2-KZeqyAAAI HTTP/1.1" 200 OK
INFO:     147.235.193.174:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHRL&sid=cOHi1ZjDd2-KZeqyAAAI HTTP/1.1" 200 OK
INFO:     ('2.53.174.122', 0) - "WebSocket /ws/socket.io/?EIO=4&transport=websocket&sid=OPGFK9cpWSPSllA0AAAE" [accepted]
INFO:     connection open
INFO:     2.53.174.122:0 - "GET /ws/socket.io/?EIO=4&transport=polling&t=P7WYHIs&sid=OPGFK9cpWSPSllA0AAAE HTTP/1.1" 200 OK`

**Screenshots/Screen Recordings (if applicable):**

![image](https://github.com/user-attachments/assets/905f99c3-471a-4081-b82e-d2d1f4fce468)

**Comment by colinatkepler** [comment]
I have this same issue, and when I query `<open-webui>/api/config`, there is no `oauth` key. I have confirmed the environment variables are set in the docker container. It appears they aren't making it through the config process.

**Reply by nick-fullpath** [reply]
Exactly! I have checked the environment variables as well in my container and they are populated correctly 
![image](https://github.com/user-attachments/assets/731cc537-1ab6-416c-9418-42bbd32e25f1)

**Reply by colinatkepler** [reply]
any new information on your side @nick-fullpath?

**Reply by nick-fullpath** [reply]
@colinatkepler Unfortunately No. 
This issue made us stop using this project.

**Comment by colinatkepler** [comment]
@nick-fullpath I upgraded my open-webui to the latest docker image and now Google Oauth works.
